### PR TITLE
HRCPP-231 fix lifespan and maxidle syntax

### PR DESCRIPTION
**Pull request by rigazilla**
please review

### CODE REVIEW

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 307](http://ci.infinispan.org/viewLog.html?buildId=36081&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 1a5194cd27bb93e2284d252e88addb67393ba2af
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:02:01

Process exited with code 1

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 440](http://ci.infinispan.org/viewLog.html?buildId=36083&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 1a5194cd27bb93e2284d252e88addb67393ba2af
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:36

Process exited with code 8

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 439](http://ci.infinispan.org/viewLog.html?buildId=36082&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 1a5194cd27bb93e2284d252e88addb67393ba2af
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:58

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 441](http://ci.infinispan.org/viewLog.html?buildId=36084&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 1a5194cd27bb93e2284d252e88addb67393ba2af
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:52

Process exited with code 8

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 442](http://ci.infinispan.org/viewLog.html?buildId=36085&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 1a5194cd27bb93e2284d252e88addb67393ba2af
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:27:51

Process exited with code 8

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 443](http://ci.infinispan.org/viewLog.html?buildId=36086&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 1a5194cd27bb93e2284d252e88addb67393ba2af
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:28:08

Process exited with code 8

File: src/hotrod/impl/TimeUnitParam.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+#include "infinispan/hotrod/TimeUnit.h"
+
+namespace infinispan {
+namespace hotrod {
+    class TimeUnitParam {
+        static TimeUnit encodeDuration(long duration, TimeUnit timeUnit) {
+            if (duration == 0) return DEFAULT;
+            if (duration < 0) return INFINITE;
+            return timeUnit;
```
Comment (tristantarrant) [inline src/hotrod/impl/TimeUnitParam.h]: You are discarding duration here...

Comment (rigazilla) [inline src/hotrod/impl/TimeUnitParam.h]: this function evaluates only the measurement unit (4 bits). The result should be the same of the java version TimeUnitParam.encodeDuration().

**Comment by tristantarrant** [comment]
Pushed, thanks

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 321](http://ci.infinispan.org/viewLog.html?buildId=36224&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 1a5194cd27bb93e2284d252e88addb67393ba2af
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:42

Process exited with code 1

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 472](http://ci.infinispan.org/viewLog.html?buildId=36233&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 1a5194cd27bb93e2284d252e88addb67393ba2af
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:07:19

Process exited with code 8

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 473](http://ci.infinispan.org/viewLog.html?buildId=36234&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 1a5194cd27bb93e2284d252e88addb67393ba2af
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:45

Process exited with code 8

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 474](http://ci.infinispan.org/viewLog.html?buildId=36235&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 1a5194cd27bb93e2284d252e88addb67393ba2af
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:28:39

Process exited with code 8

**Pull request closed without merging**